backend/src/main/java/gov/cdc/usds/simplereport/service/DataHubUploaderService.java

```diff
@@ -0,0 +1,137 @@
+package gov.cdc.usds.simplereport.service;
+
+import com.fasterxml.jackson.dataformat.csv.CsvGenerator;
+import com.fasterxml.jackson.dataformat.csv.CsvMapper;
+import com.fasterxml.jackson.dataformat.csv.CsvSchema;
+import gov.cdc.usds.simplereport.api.model.TestEventExport;
+import gov.cdc.usds.simplereport.db.model.TestEvent;
+import gov.cdc.usds.simplereport.db.model.auxiliary.RaceArrayConverter;
+import gov.cdc.usds.simplereport.db.repository.TestEventRepository;
+import net.minidev.json.JSONObject;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.web.client.RestTemplateBuilder;
+import org.springframework.core.io.ByteArrayResource;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.MediaType;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+import org.springframework.web.client.RestClientException;
+import org.springframework.web.client.RestTemplate;
+
+import javax.persistence.NoResultException;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.text.SimpleDateFormat;
+import java.time.format.DateTimeParseException;
+import java.util.ArrayList;
+import java.util.List;
+
+@Service
+@Transactional
+public class DataHubUploaderService {
+    @Value("${simple-report.data-hub.uploadurl}")
+    private String DATAHUB_UPLOAD_URL;
+    static final int MAX_ROWS_ALLOWED_PER_BATCH = 999;
+    static final String API_VERSION = "2";
+    private static final Logger LOG = LoggerFactory.getLogger(RaceArrayConverter.class);
+
+    private final TestEventRepository _repo;
+    private String _fileContents;
+    private String _nextTimestamp;
+    private String _warnMessage;
+    private String _uploadResultId;
+    private int _rowCount;
+
+    public DataHubUploaderService(TestEventRepository _repo) {
+        this._repo = _repo;
+        this._nextTimestamp = "";
```

Comment:
I would protest the use of a String for a timestamp, but I think that's probably the least of our problems... let's definitely get this stored more appropriately.

Reply:
My goal was to make the value more opaque to everything the but the database since it's only used for db compares. When it comes from the url it's a string, when it's sent back in the page results for the "next timestamp" it's a string. 

But passing it through a java datetime/instant was more opportunity to mess things up.
url (string) => java (object) => query (string) => db (timestamp) => query result (string) => java (object) => response (string).

vs

Url (string) => java (string) => query (string) => db (timestamp) => query result (string) => java (string) => response (string).

Reply:
Think this can be part of the "make it more Springy" for round 2?